src/WebJobs.Script/Diagnostics/OpenTelemetry/OpenTelemetryConfigurationExtensions.cs

```diff
@@ -42,82 +62,96 @@ internal static void ConfigureOpenTelemetry(this ILoggingBuilder loggingBuilder,
+                                       {
+                                           return false;
+                                       }
+                                   }
+
+                                   return true;
+                               };
+                           });
 
                     if (enableOtlp)
                     {
-                        b.AddOtlpExporter();
+                        builder.AddOtlpExporter();
                     }
 
-                    if (!string.IsNullOrEmpty(azMonConnectionString))
+                    if (enableAzureMonitor)
                     {
-                        b.AddAzureMonitorTraceExporter(options => ConfigureAzureMonitorOptions(options, azMonConnectionString, credential));
-                        b.AddLiveMetrics(options => ConfigureAzureMonitorOptions(options, azMonConnectionString, credential));
+                        builder.AddAzureMonitorTraceExporter(opt => ConfigureAzureMonitorOptions(opt, azMonConnectionString, credential));
+                        builder.AddLiveMetrics(opt => ConfigureAzureMonitorOptions(opt, azMonConnectionString, credential));
                     }
 
-                    b.AddProcessor(ActivitySanitizingProcessor.Instance);
-                    b.AddProcessor(TraceFilterProcessor.Instance);
+                    builder.AddProcessor(ActivitySanitizingProcessor.Instance)
+                           .AddProcessor(TraceFilterProcessor.Instance);
                 });
+        }
 
-            string eventLogLevel = GetConfigurationValue(EnvironmentSettingNames.OpenTelemetryEventListenerLogLevel, context.Configuration);
-            if (!string.IsNullOrEmpty(eventLogLevel))
-            {
-                if (Enum.TryParse(eventLogLevel, ignoreCase: true, out EventLevel level))
-                {
-                    loggingBuilder.Services.AddHostedService(service => new OpenTelemetryEventListenerService(level));
-                }
-                else
-                {
-                    throw new InvalidEnumArgumentException($"Invalid '{EnvironmentSettingNames.OpenTelemetryEventListenerLogLevel}' of '{eventLogLevel}'.");
-                }
-            }
-            else
-            {
-                // Log all warnings and above by default.
-                loggingBuilder.Services.AddHostedService(service => new OpenTelemetryEventListenerService(EventLevel.Warning));
-            }
+        private static ILoggingBuilder AddDefaultOpenTelemetryFilters(this ILoggingBuilder loggingBuilder)
+        {
+            return loggingBuilder
+                // These are messages piped back to the host from the worker - we don't handle these anymore if the worker has OpenTelemetry enabled.
+                // Instead, we expect the user's own code to be logging these where they want them to go.
+                .AddFilter<OpenTelemetryLoggerProvider>("Function.*", _ => !ScriptHost.WorkerOpenTelemetryEnabled)
```

Comment:
Curious why we explicitly care about `WorkerOptenTelemetryEnabled`? What if worker is using app-insights. Don't ask _why_ an app would use OTel in host and AppInsights in the worker, but what if they did?

What I am getting at is, should there be a general flag of worker is managing its own telemetry, and agnostic to exactly how it is doing so?

Reply:
I agree with you. We should give customers the flexibility to configure it without requiring them to use AppInsights or OTel. I'll create a work item to discuss this further.